hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBSstFileLoader.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.utils.db;
+
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.IngestExternalFileOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.apache.hadoop.hdds.utils.HddsServerUtil.toIOException;
+
+/**
+ * DumpFileLoader using rocksdb sst files.
+ */
+public class RDBSstFileLoader implements DumpFileLoader {
+
+  private final RocksDB db;
+  private final ColumnFamilyHandle handle;
+  private final IngestExternalFileOptions ingestOptions;
+
+
+  public RDBSstFileLoader(RocksDB db, ColumnFamilyHandle handle) {
+    this.db = db;
+    this.handle = handle;
+    this.ingestOptions = new IngestExternalFileOptions()
```

Comment:
https://github.com/facebook/rocksdb/wiki/Creating-and-Ingesting-SST-files

We can ingest all sst files for one container in one time, with the all-or-nothing effect.  Will this be better?

Reply:
We have multiple column families, so if the sst file carries the column family info and the ingest API knows it, then we could. But I think it doesn't.
As I understand, this ingest multiple files API targets at the scenario that there are multiple sst files that could be ingested into the same column family at once.

Reply:
Ingest API supports multiple CF，

“_DB::IngestExternalFiles() ingests a collection of external SST files for multiple column families following the 'all-or-nothing' property. If the function returns Status::OK, then all files are ingested successfully for all column families of interest. If the function returns non-OK status, then none of the files are ingested into none of the column families._”

Reply:
Unluckily, this Ingest API is not available for rocksdb java API, even with the master branch, you could check it.

Reply:
https://javadoc.io/doc/org.rocksdb/rocksdbjni/7.0.3/org/rocksdb/RocksDB.html#ingestExternalFile-java.util.List-org.rocksdb.IngestExternalFileOptions  

Does this one apply to only one CF too? 
public void ingestExternalFile(java.util.List<java.lang.String> filePathList,
                               [IngestExternalFileOptions](https://javadoc.io/static/org.rocksdb/rocksdbjni/7.0.3/org/rocksdb/IngestExternalFileOptions.html) ingestExternalFileOptions)
                        throws [RocksDBException](https://javadoc.io/static/org.rocksdb/rocksdbjni/7.0.3/org/rocksdb/RocksDBException.html)

Reply:
`
  /**
   * ingestExternalFile will load a list of external SST files (1) into the DB
   * We will try to find the lowest possible level that the file can fit in, and
   * ingest the file into this level (2). A file that have a key range that
   * overlap with the memtable key range will require us to Flush the memtable
   * first before ingesting the file.
   *
   * (1) External SST files can be created using {@link SstFileWriter}
   * (2) We will try to ingest the files to the lowest possible level
   * even if the file compression doesn't match the level compression
   *
   * @param filePathList The list of files to ingest
   * @param ingestExternalFileOptions the options for the ingestion
   *
   * @throws RocksDBException thrown if error happens in underlying
   *     native library.
   */
  public void ingestExternalFile(final List<String> filePathList,
      final IngestExternalFileOptions ingestExternalFileOptions)
      throws RocksDBException {
    ingestExternalFile(nativeHandle_, getDefaultColumnFamily().nativeHandle_,
        filePathList.toArray(new String[0]),
        filePathList.size(), ingestExternalFileOptions.nativeHandle_);
  }
`
Yes, only to the default CF.

Reply:
OK. It's a pity than FB don't expose this function in Java.